quipucords/scanner/network/inspect.py

```diff
@@ -65,7 +69,33 @@ def execute_task(self):
         failures (host/ip). Runs a host scan on the set of systems that are
         reachable. Collects the associated facts for the scanned systems
         """
-        self.connect_scan_task = self.scan_task.prerequisites.first()
+        message, status = self.check_connection()
+        if status != ScanTask.COMPLETED:
+            return message, status
+
+        message, status = self.inspect()
+        return message, status
+
+    def check_connection(self):
+        """
+        Check the connection before inspecting.
+
+        This is redundant because we could just scan immediately and handle
+        its failure as needed, but this exists due to legacy design decision
+        that requires an existing list of connection results to be referenced
+        later during the inspection. This could (should) be flattened into a
+        single operation.
+
+        TODO Remove this function when we remove connect scan tasks.
+        """
+        result_store = ConnectResultStore(self.connect_scan_task)
```

Comment:
I hate with so much prejudice `ConnectResultStore`

so convoluted and intertwined with django models. ugh. and this is the reason I still did not write a smoke test for network scans yet - patching this would be a nightmare.

However, I understand it is not the time to refactor this, so I just wanted to point out my hatred.

Reply:
I completely 💯💯💯% agree. `:sadnod:`